torch/_tensor.py

```diff
@@ -103,7 +110,24 @@ def __reduce_ex__(self, proto):
         if has_torch_function_unary(self):
             return handle_torch_function(Tensor.__reduce_ex__, (self,), self, proto)
         func, args = self._reduce_ex_internal(proto)
-        return (_rebuild_from_type, (func, type(self), args, self.__dict__))
+        # Get the state of the python subclass
+        # This loosely mimicks the function on the object class but since Tensor do not inherit
+        # from it, we cannot call that function directly
+        # https://github.com/python/cpython/blob/c83919bd635f4433f1c6ae8504996a9fe3c215e5/Objects/typeobject.c#L4891
+        if hasattr(self, "__getstate__"):
+            state = self.__getstate__()
+        else:
+            # Contrary to the cpython function, we unify dict and slots into a single dictionary
+            # instead of a tuple of two dictionaries
```

Comment:
Why? BC? Seems better to separate them. Just make a `_rebuild_from_type_v2`

Reply:
Updated